package.json

```diff
@@ -11,10 +11,11 @@
   },
   "scripts": {
     "start": "webpack serve --port 8083 --config build-tools/webpack/core-dev-server --no-client-overlay-warnings",
-    "build": "rimraf build && tsc && tsc-alias",
+    "build": "npm run core-generate-version && rimraf build && tsc && tsc-alias",
```

Comment:
wäre das postinstall script nicht eine angemessenere stelle, um das core-generate-version script auszuführen?

Reply:
Ich hoffe ich hab deine Anmerkung korrekt verstanden. Ein `npm run build` muss ich zwingend bei jeder Änderung im Core durchführen. Das script `postinstall` wird aber nur ausgeführt, wenn ich auch ein `npm i` mache, was aber nicht mit jeder Änderung am Core notwendig ist. Oder übersehe ich hier etwas?

Reply:
Meine Überlegung ist, dass sich die relevante Information (Version in der package.json) ja nur ändert/ändern kann, wenn eine installation vorgenommen wird.

Reply:
und anders herum, ein "build" script keine änderungen im source verzeichnis erzeugen sollte

Reply:
Habe es wie besprochen angepasst:

1. um in Zukunft die Version hochzuziehen muss `npm version` genutzt werden. Nur dieser löst den Hook aus zum erstellen der CoreVersion Konstante
2. es war notwendig eine neue Datei im Core zu hinterlegen: `.npmrc`. Diese deaktiviert das automatische comitten und tag erzeugen.
3. das erzeugen der Konstante erfolgt jetzt mit dem neuen script `postversion` + `core-generate-version`